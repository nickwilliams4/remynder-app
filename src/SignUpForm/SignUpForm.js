import React, { Component } from 'react'
import TokenService from '../services/token-service'
import { Button, Input, Required } from '../Utils/Utils'
import AuthApiService from '../services/auth-api-service'
import './SignUpForm.css'


export default class SignUpForm extends Component {
  static defaultProps = {
    onRegistrationSuccess: () => { }
  }
  state = { error: null }

  handleSubmit = ev => {
    ev.preventDefault()
    const { full_name, user_name, password } = ev.target

    this.setState({ error: null })
    AuthApiService.postUser({
      user_name: user_name.value,
      password: password.value,
      full_name: full_name.value
    })
      .then(user => {
        full_name.value = ''
        user_name.value = ''
        password.value = ''
        TokenService.saveAuthToken(user.authToken)
        this.context.updateLoggedIn(true)
        this.props.onRegistrationSuccess()
      })
      .catch(res => {
        this.setState({ error: res.error })
      })
  }

  render() {
    return (
      <form className='RegistrationForm'
        onSubmit={this.handleSubmit}
      >
        <div>
          <h1>Welcome!</h1>
        </div>
        <div>
          <h3>Please fill out the form below</h3>
        </div>
        <div className='full_name'>
          <label htmlFor='RegistrationForm__full_name'>Name: {' '} <Required /> </label>
          <Input
            placeholder='Name'
            type="text"
            name='full_name'
            id='RegistrationForm__full_name'>
          </Input>
        </div>
        <div className='user_name'>
          <label htmlFor='RegistrationForm__user_name'>
            Email: {' '} <Required />
          </label>
          <Input
            type="text"
            name='user_name'
            id='RegistrationForm__user_name'
            placeholder="john@lennon.com">
          </Input>
        </div>
        <div className='password'>
          <label htmlFor='RegistrationForm__password'>
            Password: {' '} <Required />
          </label>
          <Input
            name='password'
            type='password'
            required
            id='RegistrationForm__password'>
          </Input>
        </div>
        <div>*Password must be at least 8 characters, contain a capital letter, a number, and a special character.</div>
        <Button type='submit'>
          Register
        </Button>
      </form>
    )
  }
}