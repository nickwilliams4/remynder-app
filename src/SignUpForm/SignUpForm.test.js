import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SignUpForm from './SignUpForm'

describe('SignUpForm component', () => {
  const props = {
    className: 'test-class-name'
  }

  it('renders the SignUpForm given props', () => {
    const wrapper = shallow(<SignUpForm {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders a form.SignUpForm by default', () => {
    const wrapper = shallow(<SignUpForm />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})