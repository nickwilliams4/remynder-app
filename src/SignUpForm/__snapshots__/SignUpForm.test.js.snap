// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUpForm component renders a form.SignUpForm by default 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div>
    <h1>
      Welcome!
    </h1>
  </div>
  <div>
    <h3>
      Please fill out the form below
    </h3>
  </div>
  <div
    className="full-form"
  >
    <div
      className="full_name"
    >
      <label
        htmlFor="RegistrationForm__full_name"
      >
        Name: 
         
         
        <Required />
         
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__full_name"
        name="full_name"
        placeholder="Name"
        type="text"
      />
    </div>
    <div
      className="user_name"
    >
      <label
        htmlFor="RegistrationForm__user_name"
      >
        Email: 
         
         
        <Required />
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__user_name"
        name="user_name"
        placeholder="john@lennon.com"
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password: 
         
         
        <Required />
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__password"
        name="password"
        required={true}
        type="password"
      />
    </div>
     
     
    <Button
      class="new-user-login-button"
      type="submit"
    >
      Register
    </Button>
  </div>
</form>
`;

exports[`SignUpForm component renders the SignUpForm given props 1`] = `
<form
  className="RegistrationForm"
  onSubmit={[Function]}
>
  <div>
    <h1>
      Welcome!
    </h1>
  </div>
  <div>
    <h3>
      Please fill out the form below
    </h3>
  </div>
  <div
    className="full-form"
  >
    <div
      className="full_name"
    >
      <label
        htmlFor="RegistrationForm__full_name"
      >
        Name: 
         
         
        <Required />
         
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__full_name"
        name="full_name"
        placeholder="Name"
        type="text"
      />
    </div>
    <div
      className="user_name"
    >
      <label
        htmlFor="RegistrationForm__user_name"
      >
        Email: 
         
         
        <Required />
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__user_name"
        name="user_name"
        placeholder="john@lennon.com"
        type="text"
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="RegistrationForm__password"
      >
        Password: 
         
         
        <Required />
      </label>
      <Input
        class="user-name-field"
        id="RegistrationForm__password"
        name="password"
        required={true}
        type="password"
      />
    </div>
     
     
    <Button
      class="new-user-login-button"
      type="submit"
    >
      Register
    </Button>
  </div>
</form>
`;
